source/common/http/http2/codec_impl.cc

```diff
@@ -534,12 +558,19 @@ Http::Status ConnectionImpl::innerDispatch(Buffer::Instance& data) {
     dispatching_ = true;
     ssize_t rc =
         nghttp2_session_mem_recv(session_, static_cast<const uint8_t*>(slice.mem_), slice.len_);
-    if (rc == NGHTTP2_ERR_FLOODED || flood_detected_) {
-      throw FrameFloodException(
+    if (!nghttp2_callback_status_.ok()) {
+      return nghttp2_callback_status_;
+    }
+    // This error is returned when nghttp2 library detected a frame flood by one of its
+    // internal mechanisms. Most flood protection is done by Envoy's codec and this error
+    // should never be returned. However it is handled here in case nghttp2 has some flood
+    // protections that Envoy's codec does not have.
+    if (rc == NGHTTP2_ERR_FLOODED) {
```

Comment:
Does this path have coverage? If not can you add a TODO and try to add it later? In general do we have good coverage on all of the new error handling?

Reply:
No, we do not have test coverage for this case. We try to avoid tripping this case in the integration tests: https://github.com/envoyproxy/envoy/blob/dbbcc692621537d0cb4abd54950bb6356b6d01d6/test/integration/http2_integration_test.cc#L1502

It is a good question what should be done with internal nghttp2 flood protection. You could not disable it in the past (maybe you can now), so we just set the flood detection limits lower in Envoy. I could construct test for this specifically. Maybe not an integration test as it depends on being able to do a big read in one go, which is harder to control in the integration test.

No, we do not have any tests for error cases that have would end in RELEASE_ASSERT now. These would have to be added as RELEASE_ASSERTs are removed.